C#/Botkit/Microsoft.BotKit/BotkitFrameworkAdapter.cs

```diff
@@ -1,33 +1,105 @@
-// Copyright(c) Microsoft Corporation.All rights reserved.
-// Licensed under the MIT License.
-
-using Microsoft.Bot.Builder;
-using Microsoft.Bot.Connector.Authentication;
 using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
 using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Newtonsoft.Json;
 
 namespace Microsoft.BotKit
 {
     public class BotkitFrameworkAdapter : BotFrameworkAdapter
     {
+        protected readonly MicrosoftAppCredentials _credentials;
+
         /// <summary>
         /// This class extends the BotFrameworkAdapter with a few additional features to support Microsoft Teams.
         /// </summary>
         /// <param name="options"></param>
-        public BotkitFrameworkAdapter(ICredentialProvider options) : base(options)
+        /// <param name="credentials"></param>
+        public BotkitFrameworkAdapter(ICredentialProvider options, MicrosoftAppCredentials credentials) : base(options)
         {
-
+            _credentials = credentials;
         }
 
         /// <summary>
         /// Get the list of channels in a MS Teams team.
         /// Can only be called with a TurnContext that originated in a team conversation - 1:1 conversations happen _outside a team_ and thus do not contain the required information to call this API.
         /// </summary>
-        /// <param name="turnContext">A TurnContext object representing a message or event from a user in Teams</param>
-        /// <returns>An array of channels in the format [{name: string, id: string}]</returns>
+        /// <param name="turnContext">A TurnContext object representing a message or event from a user in Teams.</param>
+        /// <returns>An array of channels in the format [{name: string, id: string}].</returns>
         public async Task<Tuple<string, string>> GetChannels(TurnContext turnContext)
```

Comment:
I've compared this return type with the TS one, which is `Promise<{id: string; name: string}[]>` (check [here](https://github.com/southworkscom/botkit/blob/add/csharp-porting/packages/botkit/src/adapter.ts#L72)), and I think that the best type is an array of an _Object_ type, which contains (at least) two string properties, one for _ID_ and another for _name_, which can be translated to something like:

```
public async Task<MyType[]> GetChannels(TurnContext turnContext) ... 
```

Where _MyType_ is something like:
```
public class MyType
{
   public string Id {get;set;}
   public string Name {get;set;}
}
```